services/server/server_windows.go

```diff
@@ -20,12 +20,73 @@ package server
 
 import (
 	"context"
+	"unsafe"
 
+	"github.com/containerd/containerd/log"
 	srvconfig "github.com/containerd/containerd/services/server/config"
 	"github.com/containerd/ttrpc"
+	"github.com/pkg/errors"
+	"golang.org/x/sys/windows"
 )
 
-func apply(_ context.Context, _ *srvconfig.Config) error {
+func getPriorityClass(priorityClassName string) uint32 {
+	var priorityClassMap = map[string]uint32{
+		"IDLE_PRIORITY_CLASS":         uint32(windows.IDLE_PRIORITY_CLASS),
+		"BELOW_NORMAL_PRIORITY_CLASS": uint32(windows.BELOW_NORMAL_PRIORITY_CLASS),
+		"NORMAL_PRIORITY_CLASS":       uint32(windows.NORMAL_PRIORITY_CLASS),
+		"ABOVE_NORMAL_PRIORITY_CLASS": uint32(windows.ABOVE_NORMAL_PRIORITY_CLASS),
+		"HIGH_PRIORITY_CLASS":         uint32(windows.HIGH_PRIORITY_CLASS),
+		"REALTIME_PRIORITY_CLASS":     uint32(windows.REALTIME_PRIORITY_CLASS),
+	}
+	return priorityClassMap[priorityClassName]
+}
+
+// addCurrentProcessToJobObjectAndSetPriorityClass creates a new Job Object
+// (https://docs.microsoft.com/en-us/windows/win32/procthread/job-objects),
+// adds the current process to the job object, and specifies the priority
+// class for the job object to the specified value.
+// A job object is used here so that any spawned processes such as CNI or
+// shim binaries are created at the specified thread priority class.
+// Running containerd and shim binaries with above normal / high priority
+// can help improve responsiveness on machines with high CPU utilization.
+func addCurrentProcessToJobObjectAndSetPriorityClass(pc uint32) error {
+	job, err := windows.CreateJobObject(nil, nil)
+	if err != nil {
+		return errors.Wrap(err, "windows.CreateJobObject failed")
+	}
+	limitInfo := windows.JOBOBJECT_BASIC_LIMIT_INFORMATION{
+		LimitFlags:    windows.JOB_OBJECT_LIMIT_PRIORITY_CLASS,
+		PriorityClass: pc,
+	}
+	if _, err := windows.SetInformationJobObject(
+		job,
+		windows.JobObjectBasicLimitInformation,
+		uintptr(unsafe.Pointer(&limitInfo)),
+		uint32(unsafe.Sizeof(limitInfo))); err != nil {
+		return errors.Wrap(err, "windows.SetInformationJobObject failed")
+	}
+	if err := windows.AssignProcessToJobObject(job, windows.CurrentProcess()); err != nil {
+		return errors.Wrap(err, "windows.AssignProcessToJobObject failed")
+	}
+	return nil
+}
+
+func apply(ctx context.Context, config *srvconfig.Config) error {
+	priorityClassName := config.OS.Windows.PriorityClass
+	if priorityClassName != "" {
+		log.G(ctx).Infof("Setting process priority class to %s", priorityClassName)
+
+		pc := getPriorityClass(priorityClassName)
+		if pc == 0 {
+			return errors.Errorf("Invalid priority class %s, valid priority classes are defined at "+
+				"https://docs.microsoft.com/en-us/windows/win32/procthread/scheduling-priorities", priorityClassName)
+		}
+		if err := addCurrentProcessToJobObjectAndSetPriorityClass(pc); err != nil {
+			log.G(ctx).Error(err)
```

Comment:
Seems odd to log this if we're just going to return the err (and abort the program if this failed) also